File: tests/java/com/trulioo/sdk/model/DataFieldTest.java
package com.trulioo.sdk.model;

import static org.junit.jupiter.api.Assertions.*;

import org.junit.jupiter.api.*;

public class DataFieldTest {
    private DataField dataField;

    @BeforeEach
    public void beforeEach() {
        dataField = new DataField();
    }

    @Test
    public void testFieldName() {
        String fieldName = "field";
        dataField.setFieldName(fieldName);

        assertEquals(fieldName, dataField.getFieldName());
        assertEquals(fieldName, new DataField().fieldName(fieldName).getFieldName());
    }

    @Test
    public void testValue() {
        String value = "test";
        dataField.setValue(value);

        assertEquals(value, dataField.getValue());
        assertEquals(value, new DataField().value(value).getValue());
    }

    @Test
    public void testFieldGroup() {
        String fieldGroup = "group";
        dataField.setFieldGroup(fieldGroup);

        assertEquals(fieldGroup, dataField.getFieldGroup());
        assertEquals(fieldGroup, new DataField().fieldGroup(fieldGroup).getFieldGroup());
    }

    @Test
    public void testEquals() {
        String fieldName = "field";
        DataField dataField1 = new DataField().fieldName(fieldName);

        assertEquals(dataField1, dataField1);
        assertEquals(dataField1, new DataField().fieldName(fieldName));
        assertNotEquals(dataField1, new DataField().fieldName(fieldName + "1"));
        assertNotEquals(dataField1, null);
    }

    @Test
    public void testHashcode() {
        String fieldName = "field";

        assertEquals(new DataField().fieldName(fieldName).hashCode(), new DataField().fieldName(fieldName).hashCode());
    }

    @Test
    public void testToString() {
        String fieldName = "field";

        assertEquals(new DataField().fieldName(fieldName).toString(), new DataField().fieldName(fieldName).toString());
    }
}

File: tests/php/Model/DataFieldTest.php
<?php

namespace Trulioo\SDK\Test\Model;

use PHPUnit\Framework\TestCase;
use Trulioo\SDK\Model\DataField;

class DataFieldTest extends TestCase
{
    private DataField $dataField;

    public function setUp(): void
    {
        $this->dataField = new DataField([]);
    }

    public function testPropertyFieldName()
    {
        $this->dataField->setFieldName('field');
        $this->assertEquals('field', $this->dataField->getFieldName());
    }

    public function testPropertyValue()
    {
        $this->dataField->setValue('value');
        $this->assertEquals('value', $this->dataField->getValue());
    }

    public function testPropertyFieldGroup()
    {
        $this->dataField->setFieldGroup('field_group');
        $this->assertEquals('field_group', $this->dataField->getFieldGroup());
    }
}
